### DISCUSSION TITLE
Chapter 5: Adding a Cut Scene and Handling Input

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#128] [Going the Distance Discussion]
Discussion thread for Chapter 5. 

- Space-Truckers: The Splash Screen 
- Designing the input system
- Extended Topics

**Comment by AlexanderPolenov** [comment]
1. Regarding Splash Screen: https://playground.babylonjs.com/#DSALXR
Line 173 makes no sense, because `prior` is never assigned. Thus, in the example from the book, we call `this.currentSegment?.start();` on every frame (164 per second on my PC). The only reason why it works is because BabylonJS doesn't restart the playing animation on each `AnimationGroup.start` call. I honestly don't understand why we need `this.onUpdate` function, a potentially dangerous subscription that we can forget to unsubscribe. Why can't we just call `start` inside of `onEnd` handler of each segment? We are switching segments there anyway.

2. Regarding Gamepad: https://playground.babylonjs.com/#78MJJ8#64
Crashes immediately on Line 262 (function name error mapStickRotationInputToActions). Even after fixing the name the solution is far from working. Click on 'A' (button1) works, click on everything else crashes immediately. Sticks also don't work and the `enableGamepad` function doesn't seem to set up everything properly. At the same time examples from BabylonJS doc are working. So, it's not the issue on my side. Maybe it's because browsers APIs and BabylonJS APIs could change since this chapter was written. I think the whole gamepad section in the book should be rewritten. Honestly, I don't know how to fix this playground without rewriting almost everything from scratch. But in its current state this section of the book is pretty much useless.

**Reply by jelster** [reply]
Thanks for the feedback. 

> Maybe it's because browsers APIs and BabylonJS APIs could change since this chapter was written

Exactly this. The different pointer, pad, and touch API's have been rewritten a few times since the chapter was written, and even during writing it was a moving target. 

Although I've been tracking this and other feedback here and in the Issues, it's not always the most visible to a casual user. Obviously, I don't have an easy way to update the text of the book (although it is potentially doable), so what do you think might be a good way to make visible and help readers navigate their way around these types of situations?

If there's ever a second edition of the book, I'll probably end up reworking large chunks of the chapter - Coroutines would make more sense to manage the animation timing, but they also were not available in BJS at the time the chapter was written!
